osh/worker/csmock_runner.py

```diff
@@ -151,6 +166,15 @@ def srpm_download_analyze(self, analyzers, srpm_name, srpm_url, profile=None,
         srpm_path = self.download_file(srpm_url, srpm_name)
         return self.analyze(analyzers, srpm_path, profile, su_user, additional_arguments, **kwargs)
 
+    def dist_git_url_analyze(self, analyzers, dist_git_url, profile=None,
+                             su_user=None, additional_arguments=None, **kwargs):
+        if profile != "cspodman":
```

Comment:
it won't:
```
Traceback (most recent call last):
  ...
  File "/usr/lib/python3.6/site-packages/osh/hub/scan/scanner.py", line 452, in __init__
    self.validate_options()
  File "/usr/lib/python3.6/site-packages/osh/hub/scan/scanner.py", line 489, in validate_options
    raise RuntimeError("'auto' mock config is only compatible with '--nvr'")
RuntimeError: 'auto' mock config is only compatible with '--nvr'
```

Is this the intended behavior?

Reply:
It would be better to make `osh-cli` accept `--config=auto` with `--git-url`, too.  Given the fact, that `--git-url` works for container builds only, we can always translate `auto` to `cspodman` like we do for Brew builds:
- https://github.com/openscanhub/openscanhub/blob/ee9fd3ed9fbb75b2825b62bbf5edc103996aebc3/osh/hub/scan/scanner.py#L156
- https://github.com/openscanhub/openscanhub/blob/ee9fd3ed9fbb75b2825b62bbf5edc103996aebc3/osh/hub/scan/scanner.py#L246

The long term plan is to make `osh-cli` use `--config=auto` (instead of `fedora-rawhide-x86_64`) by default so that users do not need to use the option in 99% of cases.

Reply:
I've addressed it in https://github.com/openscanhub/openscanhub/pull/278/commits/0c8d6e52a06a970d21ff4a072dde079714797791 . had some tests on stage, works fine with `--config=auto`.

However this change only impacts `ClientScanScheduler`. The original jira issue doens't mention diff-build or tasks from errata, thus the PR doesn't contain support for `git-url` with diff-build. I had a quick check, to support that, there'll a bit more clumsy logic needed.

Let me know if it's needed.

Reply:
Errata Tool scans Brew builds only.  The support for `version-diff-build` tasks can be introduced later if needed.